mlflow/types/utils.py

```diff
@@ -136,6 +141,15 @@ def _infer_schema(data: Any) -> Schema:
                 for field in data.schema.fields
             ]
         )
+    elif isinstance(data, str):
+        schema = Schema([ColSpec(type=DataType.string)])
+    elif isinstance(data, list):
+        if all(isinstance(element, str) for element in data):
+            schema = Schema([ColSpec(type=DataType.string)])
+        elif all(isinstance(element, dict) for element in data) and all(
+            isinstance(value, str) for d in data for value in d.values()
+        ):
+            schema = Schema([ColSpec(type=DataType.string, name=name) for name in data[0].keys()])
```

Comment:
I think we could support `List[Dict[str, <any scalar>]]` by doing `pd.DataFrame.from_records()` on the list of dicts and then using the logic for Pandas DataFrame schema inference above:

````
    elif isinstance(data, pd.DataFrame):
        schema = Schema(
            [ColSpec(type=_infer_pandas_column(data[col]), name=col) for col in data.columns]
        )
```

Reply:
Are there any HF language model types where users need to pass dicts with non-string scalar values (e.g. int values, float values, etc), or do string values cover all cases?

Reply:
No LLMs require this (but the multimodal ones will in the future). Should we implement this when we work on functionality to support those later?